pkg/k8s/k8s_networkpolicy.go

```diff
@@ -86,3 +86,29 @@ func DeleteNetworkPolicy(namespace string, name string) error {
 
 	return nil
 }
+
+// GetNetworkPolicy retrieves a NetworkPolicy resource.
+func GetNetworkPolicy(namespace string, name string) (*v1.NetworkPolicy, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	if !IsInitialized() {
+		return nil, fmt.Errorf("knuu is not initialized")
+	}
+	np, err := Clientset().NetworkingV1().NetworkPolicies(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error getting network policy %s: %w", name, err)
+	}
+
+	return np, nil
+}
+
+// NetworkPolicyExists checks if a NetworkPolicy resource exists.
+func NetworkPolicyExists(namespace string, name string) bool {
+	_, err := GetNetworkPolicy(namespace, name)
+	if err != nil {
+		return false
+	}
```

Comment:
[non-blocking]: I would put some debug logs here on what the error msg was

Reply:
https://github.com/celestiaorg/knuu/pull/168/commits/ad4a8e6024de48d1f4b4ae0450f5b9377e345784